### PR TITLE
Parse text

### PR DESCRIPTION
**Pull request by labouz**
needs tests

### CODE REVIEW

**Comment by labouz** [comment]
@nathan-castor this function needs a test pls

**Comment by nathan-castor** [comment]
@labouz 
Looks like there are some conflicts with this code now, sorry I would've done this last week if I could have.

File: R/parse.R [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  # })
+
+  #function to subset data
+  # get_col <- function(){
+  #   prompt <- stringr::str_c(c("Which column contains drug names?:",colOptions),
+  #                            collapse = " \n")
+  #   colNum <- as.numeric(readline(prompt))
+  #   drug_data[ ,colNum] #returns tibble of n X 1
+  # }
+
```
Comment (nathan-castor) [inline R/parse.R]: Let's remove the commented code that's not serving a purpose

File: R/parse.R [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#' @return A n x 1 dataframe with class \code{data.frame}, \code{tbl_df}, \code{tbl}.
+#'   The dataframe has 1 variable, "drug"
+#' @importFrom utils data
+#' @importFrom stats na.omit
+#' @importFrom magrittr %>%
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' parse(drug_df)
```
Comment (nathan-castor) [inline R/parse.R]: Incorrect argument name (drug_vec)

File: R/parse.R [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+#' }
+
+parse <- function(drug_vec){
+
+  #binding vars to function
+  word <- drug <- for_token <- stop_words <- NULL
```
Comment (nathan-castor) [inline R/parse.R]: we don't need stop_words set to NULL here

File: R/parse.R [hunk 4]
```diff
@@ -0,0 +1,109 @@
+
+#' Parse a vector of free text containing drug information
+#'
+#' @description This function provides a dataframe of parsed out strings from a
+#'   free text field given a file path specified by the user.
```
Comment (nathan-castor) [inline R/parse.R]: edit description: we don't take a file path anymore

File: R/parse.R [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                       "injection", "er", "medication", "mgs", "illicit", "iv", "left",
+                       "patches", "visit", "hcl", "plus", "hd", "bit", "cit",
+                       "sulf", "tart", "c-ject", "es", "hp", "syringe", "contin",
+                       "intensol", "ject", "calcium")
+
+  ##INTERACTIVE - must respond!
+  # drugs <- get_col()
+  #convert vector to df
+  drugs <- as.data.frame(drug_vec)
+  drugCol <- as.name(names(drugs))
```
Comment (nathan-castor) [inline R/parse.R]: unnecessary to create this variable because you are piping the df into the function on line 67

**Review by nathan-castor — CHANGES_REQUESTED** [review]
Woohoo!

File: R/parse.R [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  drug_stop_words <- c("a", "few", "mg", "pills", "pill","days", "off", "bunch", "street",
+                       "tab", "tabs", "detox", "rx", "not", "unsure", "unknown",
+                       "clinic", "bottle", "unknkwn", "type", "patch", "pm", "which",
+                       "injection", "er", "medication", "mgs", "illicit", "iv", "left",
+                       "patches", "visit", "hcl", "plus", "hd", "bit", "cit",
+                       "sulf", "tart", "c-ject", "es", "hp", "syringe", "contin",
+                       "intensol", "ject", "calcium")
+
+  ##INTERACTIVE - must respond!
+  # drugs <- get_col()
```
Comment (nathan-castor) [inline R/parse.R]: unneeded comments (commented code)

File: R/parse.R [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  #convert vector to df
+  drugs <- as.data.frame(drug_vec)
+  # drugCol <- as.name(names(drugs))
```
Comment (nathan-castor) [inline R/parse.R]: unneeded comments (commented code)

**Review by nathan-castor — CHANGES_REQUESTED** [review]
just two lines of commented code

**Review by nathan-castor — APPROVED** [review]

**Pull request merged**